### PR TITLE
test(TextField): Jestify and Enzymify TextField component tests

### PR DESCRIPTION
**Pull request by kosanna**
No tests lost, cleanup, one step forward to deprecation of some dependencies

### CODE REVIEW

File: react/TextField/TextField.test.js [hunk 1]
```diff
@@ -1,150 +1,96 @@
...
-      expect(errors[0]).to.match(/`inputProps.id` will be overridden by `id`/);
+      shallow(<TextField inputProps="hey" />);
+      expect(spy).toBeCalledWith(expectedError);
     });
 
-    it('should pass through className to the input', () => {
-      render(<TextField inputProps={{ className: 'first-name-field' }} />);
-      expect(input.props.className).to.match(/first-name-field$/);
-    });
+    it('should error if if `inputProps`\'s `id` is specifiedt', () => {
```
Comment (markdalgleish) [inline react/TextField/TextField.test.js]: Double "if" here 😄

Comment (markdalgleish) [reply react/TextField/TextField.test.js]: Also "specifiedt".

Comment (kosanna) [reply react/TextField/TextField.test.js]: 🤦‍♂️  lol, will fix it :D

Comment (kosanna) [reply react/TextField/TextField.test.js]: Done 👍

**Pull request merged**